oidc-authentication.md

```diff
@@ -63,6 +63,183 @@
+others) to perform [authentication].
+
+The end result of any successful authentication flow is that the client
+performing it receives one or more credentials from the [identity provider]. In
+the case of classic OpenID Connect flows, those credentials are typically either
+the [id token], an [access token] specific to a particular resource server, or
+both.
+
+In addition, OpenID Connect provides a way for a [relying party] to request
+arbitrary [claims] or [credentials] inside the ID Token. In particular, this
+specification uses that mechanism to allow a  [relying party] to request one or
+more reusable credentials that enable audience-constrained [protected resource
+requests] to multiple unaffiliated [resource servers] (see section [Multi
+Resource Server (Multi-RS) Use Case] for more details).
+
+##### Web ID
+A globally unique [identifier] (typically an HTTPS URL) for an [entity], that
+can be resolved to a Web ID Profile document.
+
+##### user-agent
+As defined by RFC 2616 [RFC2616]. Often, a formal name for a web browser. Note
+that this is often separate from a [client] application (in many cases, client
+apps written in Javascript run inside the browser).
+
+##### resource controller
+An [entity] with administrative permissions to a resource (e.g. `acl:Control`
+access mode in the case of [WAC]). Alternatively, the [entity] in control of the
+[resource server] (such as a [Solid pod] or storage space) in which the resource
+resides.
+
+Similar to the OAuth 2 and
+[UMA 2](https://docs.kantarainitiative.org/uma/wg/rec-oauth-uma-grant-2.0.html)
+concept of "resource owner", the resource controller MAY be an [end-user]
+(natural person) or MAY be a non-human entity treated as a person for limited
+legal purposes (legal person), such as a corporation.
+
+Note the use of "controller" as opposed to "owner"; this specification (and
+Solid in general), has no notion of owner.
+
+A given resource can have zero, one, or multiple resource controllers.
+
+##### protected resource
+A resource to which access is constrained by an appropriate authorization
+mechanism (such as Web Access Control [WAC] rules or Authorization Capabilities
+[zCAP]).
+
+##### client
+An application, service, bot, or other software entity interacting with a
+protected resource hosted by a [resource server] on behalf of a [requesting
+party]. The term "client" does not imply any particular implementation
+characteristics (such as whether the application executes on a server, a
+desktop, inside a web browser, or on other devices).
+
+The client is an [entity] in its own right, possessing an [identifier], and in
+some cases, a corresponding [controller document] containing client metadata.
+
+##### requesting party
+An [entity] (natural or legal person) that uses a [client] to seek access to a
+[protected resource]. The requesting party may or may not be the same party as
+the [resource controller].
+
+##### Solid pod (personal online datastore)
```

Comment:
:-1: to pod combining storage and OP. Instead I consider it clearer to use term *Solid storage*. Which usually acts as OAuth2 RS and in case where accessed directly with user agent (without client application) #31 it acts as OIDC RP. Giving any OP expectations on the pod may lead to confusions. At the same time nothing prevents implementations from combining solid storage and OIDC OP together.

Reply:
Some prior conversation in https://github.com/solid/specification/issues/16 . Something one may call in non-normative way *solid pod* could also provide way to deploy apps and bots. I think we should leave that term non-normative and not constrained to storage + OP composition.

![solid](https://user-images.githubusercontent.com/876431/62135152-53663300-b2a7-11e9-9098-007f30e4371c.png)

Reply:
I don't believe that there has yet been finalization of terminology in the specification repo, but once there is, we should go with that.

At the moment I would consider `Solid Pod` to be synonomous with `Solid Storage`, so a Solid Pod would not include the identity provider component.

Reply:
Maybe we could try to agree on how we call the thing referenced by `solid:storage` ? `Solid Storage` sounds better to me, especially if we want to use `Solid Pod` as more catch all term which can compose `Solid Storage`, `Solid IdP`, and possibly other responsibilities like host for apps / bots.

Reply:
I'd be okay with just calling it Solid Storage

Reply:
Thanks. Renamed 'solid pod' term to 'solid storage'.